### ISSUE TITLE
N4JS allows private abstract methods and forces overriding them in sub classes

### ISSUE THREAD

**Issue by yoosiba** [eclipse/n4js#1047]
N4JS allows private abstract methods and forces overriding them in sub classes
```typescript
export public abstract class Abstract {
    // XPECT error ---> "abstract not allowed on private methods"
    private abstract m();
}

export public class Concrete extends Abstract {
    //XPECT error ---> "cannot override private method"
    @Override
    private m(){};
}
```
see error when `private abstract` is not overridden
![override_private](https://user-images.githubusercontent.com/944287/43892129-6de639f8-9bcb-11e8-8751-efa58a90c598.gif)

**Comment by kduske-n4** [comment]
Interestingly, making virtual (i.e. overridable) methods private is a recommended practice for C++: https://stackoverflow.com/questions/2170688/private-virtual-method-in-c

Not saying N4JS should have this, since it doesn't really make sense in a language that follows Java's footsteps. Just a point of interest.

Furthermore, whoever fixes this should check why the access control test suite didn't catch this. It's possible that it's missing the case of overriding an abstract method.

**Comment by jpilgrim** [comment]
Remark: added new constraint in spec (Req IDE-46).

Did not remove consequential error in subtype overriding the "private abstract" method -- if you  get that error, that means you have ignored the error in the super type.

**Comment by jpilgrim** [comment]
I think the quickfix does not make much sense, that's true. Again I thing we should fix that in the context of #1221.
